in case client send corrupt data

kunci115 opened the pull request:
feat: Add comprehensive data integrity verification system with client rejection

  - Implement checksum-based verification for WebSocket audio transmission
  - Add server-side rejection policies for corrupted data clients
  - Create configurable corruption thresholds (0=strict, N=tolerant)
  - Add per-client corruption tracking with automatic cleanup

  Server Features:
  - New CLI flags: --verify-data-integrity, --reject-corrupted-data, --corruption-threshold
  - Real-time verification of audio length and checksum
  - Structured rejection messages sent to clients before disconnection
  - Extended logging support for verification results

kunci115 committed:
[add] data integrity check, (mitigation of corrupt)

kunci115 committed:
[add] md for summary

kunci115 committed:
[add] sample client data integrity check

KoljaB commented:
That's a great PR. I think probably it needs some changes to stt_cli_client.py in RealtimeSTT_server folder and then maybe to audio_input.py and audio_recorder_client.py in RealtimeSTT folder to adapt to this new chunk verification logic. Since I'm doing some refacturing currently too I think I'll merge it later and then add the needed changes.

kunci115 commented:
I'll help in case I'm free on weekend if you haven't finished it, I also forgot to push the other test that I wrote

kunci115 committed:
[add] test

kunci115 committed:
Changes Made

  1. RealtimeSTT/audio_recorder_client.py

  - Added calculate_checksum() method for data verification
  - Updated feed_audio() method to include verification data when server_sent_to_stt is present
  - Added enable_data_verification parameter to constructor
  - Updated record_and_send_audio() method to include verification metadata when enabled

  2. RealtimeSTT_server/stt_cli_client.py

  - Added --verify-data command line flag
  - Updated client instantiation to pass the verification flag

  Key Features Added

  Optional Data Verification:
  - Backward compatible - verification is off by default
  - When enabled, adds checksum, data length, and timestamp to metadata
  - Follows the same pattern as the sample Python client

  CLI Integration:
  - New --verify-data flag enables verification in the CLI client
  - Usage: stt-cli-client --verify-data

  Automatic Verification:
  - feed_audio() automatically enables verification when server_sent_to_stt is in metadata
  - record_and_send_audio() includes verification when enable_data_verification=True

  Usage Examples

  # CLI with verification enabled
  stt-cli-client --verify-data --language en

  # Python code with verification
  client = AudioToTextRecorderClient(
      enable_data_verification=True,
      language="en"
  )

  The implementation maintains full backward compatibility while adding robust data integrity verification capabilities that integrate seamlessly with the existing
  server verification system.

kunci115 committed:
1. tests/test_quick_performance.py - Simple & Fast

  Usage:
  # Test your current server
  python tests/test_quick_performance.py

  # Compare default vs optimized (manual server restart)
  python tests/test_quick_performance.py --compare

  2. tests/test_performance_benchmark.py - Comprehensive

  Usage:
  # Basic benchmark
  python tests/test_performance_benchmark.py

  # Test with concurrent requests
  python tests/test_performance_benchmark.py --concurrent 5

  # Auto-start optimized server and test
  python tests/test_performance_benchmark.py --config optimized --auto-server

kunci115 committed:
[edit]

cesinsingapore committed:
[delete] wrong patch on test

kunci115 committed:
[delete] wrong patch test

cesinsingapore committed:
Merge branch 'master' of https://github.com/kunci115/RealtimeSTT

cesinsingapore committed:
[edit] bug fix for enable realtime transcription and silero deactivity detection param

cesinsingapore committed:
[edit] change to 0.0.0.0
